### PR TITLE
Spaces API Documentation Update

### PR DESCRIPTION
**Pull request by tr0njavolta**
This PR attempts to restructure the current spaces API reference documentation. Should more closely resemble the Crossplane API docs for readability and organization https://docs.crossplane.io/master/api/

To Do
- [x] Make sure the structure of content in the yaml files is correct @haarchri - would be helpful to take a look.
- [x] Color correction
- [x] UI correction

### CODE REVIEW

**Comment by haarchri** [comment]
for me the page looks empty 

image

File: themes/upbound/assets/scss/_color-mode-dark.scss [hunk 1]
```diff
@@ -64,6 +64,32 @@
     --search-box-background: var(--body-background);
     --result-path: var(--body-background);
 
+    // API Page
+    --sort-up: url("/img/sort-up-light.svg");
+    --sort-down: url("/img/sort-down-light.svg");
+    --expand-box-background: #{$up-grey-800n};
+    --minus-svg: url("/img/minus-white.svg");
+    --plus-svg: url("/img/plus-150.svg");
```
Comment (plumbis) [inline themes/upbound/assets/scss/_color-mode-dark.scss:72]: I think this needs to be `img/lightbox/...`, but these are only dark +/-. 

You can get the originals from here:
https://github.com/crossplane/docs/tree/master/themes/geekboot/static/img

Be sure to add the same vars into the `_color-mode-light.scss` file as well.

File: themes/upbound/assets/scss/_crds.scss [hunk 2]
```diff
@@ -0,0 +1,268 @@
+@mixin vertical-bar {
```
Comment (plumbis) [inline themes/upbound/assets/scss/_crds.scss:1]: In order to expand the central display area over the column where the Table of Contents normally is you have to adjust the grid layout. My suggestion is something like:

1. Above this create a new class:
```css
.crd-grid {

  @include media-breakpoint-up(xl) {
  display: grid;
  grid-template-columns: 336px minmax(0,.98fr);
  grid-template-rows: 1fr auto;
  grid-template-areas:
  "left-sidebar-nav content-body right-sidebar-toc"
  "footer-container footer-container footer-container";
  }

}
```

2. Edit `themes/upbound/layouts/_default/baseof.html` and conditionally load the grid:
```diff


    {{ partial "body/top-nav" .}}
-    <div class="grid" data-bs-spy="scroll" data-bs-target="#TableOfContents">
+    <div class="{{ if eq $.Params.Product "crds" }}crd-grid{{else}} grid {{ end }} " data-bs-spy="scroll" data-bs-target="#TableOfContents">
      {{ partialCached "icons/sprites" . }}
      {{ partial "body/left-sidebar-nav" .}}

```

**Review by plumbis — CHANGES_REQUESTED** [review]
I added some suggestions to help fix the big problems- with those in place it looks pretty good! just need to fix some coloring around the object types and nudge spacing after more screen real estate is freed up with the grid fix.

**Comment by tr0njavolta** [comment]
#330 
Should be rendering correctly now. Will work on the CSS issues now.

File: content/reference/spaceapi/yaml/sharedsecretstore.yaml [hunk 3]
```diff
@@ -0,0 +1,166 @@
+apiVersion: spaces.upbound.io/v1beta1
```
Comment (haarchri) [inline content/reference/spaceapi/yaml/sharedsecretstore.yaml:1]: i wonder what these files are - this is not a real CRD - so how you generate these files ?

File: content/reference/spaceapi/yaml/sharedsecret.yaml [hunk 4]
```diff
@@ -0,0 +1,131 @@
+apiVersion: spaces.upbound.io/v1beta1
```
Comment (haarchri) [inline content/reference/spaceapi/yaml/sharedsecret.yaml:1]: i wonder what these files are - this is not a real CRD - so how you generate these files ?

File: content/reference/spaceapi/yaml/controlplane.yaml [hunk 5]
```diff
@@ -0,0 +1,99 @@
+apiVersion: spaces.upbound.io/v1beta1
```
Comment (haarchri) [inline content/reference/spaceapi/yaml/controlplane.yaml:1]: i wonder what these files are - this is not a real CRD - so how you generate these files ?

**Comment by cwilhit** [comment]
In light mode, the connective bars also render in white so they look like they're not there:

image

**Review by plumbis — APPROVED** [review]

**Pull request merged**